### PR TITLE
Created public functions re pp archival to iRODS

### PR DESCRIPTION
**Pull request by mgcam**

### CODE REVIEW

File: lib/npg_pipeline/product/release/irods.pm [hunk 1]
```diff
@@ -15,6 +16,8 @@ our $VERSION = '0';
 Readonly::Scalar my $THOUSAND                    => 1000;
 Readonly::Scalar my $PRODUCTION_IRODS_ROOT       => q[/seq];
 Readonly::Scalar my $IRODS_REL_ROOT_NOVASEQ_RUNS => q[illumina/runs];
+Readonly::Scalar my $IRODS_PP_CONF_KEY           =>
```
Comment (nerdstrike) [inline lib/npg_pipeline/product/release/irods.pm:19]: Not sure how I feel about aliasing a constant. Does the composed `release` role not import this constant into the namespace?

File: lib/npg_pipeline/product/release/irods.pm [hunk 2]
```diff
@@ -194,10 +197,60 @@ sub is_for_irods_release {
...
+'exclude' filter is optional.
+
+The format for both filters is validated. It is expected that an array of
+filter expressions is present.
+
+  my $filters = $self->glob_filters4publisher($product)
+
+=cut
+
+sub glob_filters4publisher {
```
Comment (nerdstrike) [inline lib/npg_pipeline/product/release/irods.pm:229]: I'm loath to point out that this can now be tested independently.

**Review by nerdstrike — CHANGES_REQUESTED** [review]
One typo to fix, and a couple of questions I'd like to see addressed.

Tests run without incident

File: Changes [hunk 3]
```diff
@@ -1,6 +1,10 @@
 LIST OF CHANGES
 ---------------
 
+ - Refactored existing code to create new public functions, which will be
+   used in the extention of npg_run_deletable script dealing with decisions
```
[the remark this answers is not in the document]
Comment (mgcam) [reply Changes]: Thanks, fixed.

File: lib/npg_pipeline/product/release/irods.pm (hunk 2 above)
Comment (mgcam) [reply lib/npg_pipeline/product/release/irods.pm:229]: Agree

File: t/15-product-release-irods.t [hunk 4]
```diff
@@ -1,42 +1,94 @@
...
+    lims => st::api::lims->new(id_run => 34576, position => 1, tag_index =>3));
+  my $class = Moose::Meta::Class->create_anon_class(roles => [$package]);
+
+  my $obj = $class->new_object(product_conf_file_path =>
+    q[t/data/release/config/pp_archival/product_release.yml]);
+  ok ($obj->is_for_pp_irods_release($product),
+    'this product should be in the pp iRODS archive');
+  my $filters = $obj->glob_filters4publisher($product);
+  ok ($filters, 'filters are returned');
+  is (keys %{$filters}, 2,'two types of filters');
```
Comment (nerdstrike) [inline t/15-product-release-irods.t:64]: Comparing two integers by casting to string, but I'll allow it.

**Review by nerdstrike — APPROVED** [review]

**Pull request merged**